src/main/java/game/Main.java

```diff
@@ -4,31 +4,40 @@
 import interfaces.GameFlow;
 import interfaces.InputHash;
 import map.BattleMap;
+import Hinter.Hints;
 
 import java.util.Scanner;
 
 public class Main {
 
-    public static boolean devMode = false;
+    public static boolean devMode = true;
 //    public static InputValidator validator;
     public static GameFlow gameFlow = new GameFlowImpl(devMode);
     public static InputHash inputHash = new InputHashImpl();
 
+
+
     public static void main(String[] args) throws InterruptedException {
+        Hints nextHint = new Hints();
         BattleMap battleMap = new BattleMap(4, 2);
         Scanner scanner = new Scanner(System.in);
 
         gameFlow.showIntro();
 
         while (battleMap.hasShips()) {
-            String attackKey = askForInput(scanner);
 
+            String attackKey = askForInput(scanner);
+            if(attackKey.equalsIgnoreCase("hint")) {
+               System.out.println(nextHint.printNextHint());
+               System.out.println("You have " + nextHint.getSize() + " hints left.\n");
+            }
+            else {
+                boolean isSuccessfulAttack = attack(battleMap, attackKey);
+                gameFlow.showAttackResult(isSuccessfulAttack);
+            }
```

Comment:
Instead of an if-else we can do if -> continue. And leave the rest as is

```suggestion
            if(attackKey.equalsIgnoreCase("hint")) {
               System.out.println(nextHint.printNextHint());
               System.out.println("You have " + nextHint.getSize() + " hints left.\n");
               continue;
            }

            boolean isSuccessfulAttack = attack(battleMap, attackKey);
            gameFlow.showAttackResult(isSuccessfulAttack);
           
```